File: nexobytes_edutech/lib/classes/chapter.dart
class Chapter {
  String id;
  String name;
  int createdAt;
  String createdBy;
  String classId;
  String className;
  String subjectId;
  String subjectName;

  Chapter({
    required this.id,
    required this.name,
    this.createdAt=0,
    this.createdBy="",
    this.classId = "",
    this.className="",
    this.subjectId="",
    this.subjectName=""
  });
}

File: tempserver/models/chapter.js
const mongoose = require("mongoose");
const uniqueValidator = require('mongoose-unique-validator');

const Schema = mongoose.Schema;

const chapterSchema = new Schema({
	chapterId: { type: String, required: true, unique: true },
	classId: { type: String, required: true },
	className: { type: String, required: true },
	createdAt: { type: Number, required: true },
	createdBy: { type: String, required: true },
	name: { type: String, required: true },
	subjectId: { type: String, required: true },
	subjectName: { type: String, required: true },
});

chapterSchema.plugin(uniqueValidator);

module.exports = mongoose.model("Chapter", chapterSchema);
